src/drivers/imx/dummy-dma.c

```diff
@@ -3,38 +3,159 @@
 // Copyright 2019 NXP
 //
 // Author: Daniel Baluta <daniel.baluta@nxp.com>
+// Author: Paul Olaru <paul.olaru@nxp.com>
 
+/* Dummy DMA driver (software-based DMA controller)
+ *
+ * This driver is usable on all platforms where the DSP can directly access
+ * all of the host physical memory (or at least the host buffers).
+ *
+ * The way this driver works is that it simply performs the copies
+ * synchronously within the dma_start() and dma_copy() calls.
+ *
+ * One of the drawbacks of this driver is that it doesn't actually have a true
+ * IRQ context, as the copy is done synchronously and the IRQ callbacks are
+ * called in process context.
+ *
+ * An actual hardware DMA driver may be preferable because of the above
+ * drawback which comes from a software implementation. But if there isn't any
+ * hardware DMA controller dedicated for the host this driver can be used.
+ *
+ * This driver requires physical addresses in the elems. This assumption only
+ * holds if you have CONFIG_HOST_PTABLE enabled, at least currently.
+ */
+
+#include <ipc/topology.h>
```

Comment:
Goes after `#include <sof/trace/trace.h>` and before `#include <user/trace.h>`.

Reply:
So it's sof, ipc, user? Will do, I still don't fully grasp the ordering.